File: 241005/컴퓨터와 함께하는 숫자 게임 2/play-number-game-with-computer-2.py
def game_duration(m, a, b):
    # 최소 횟수와 최대 횟수를 초기화합니다.
    min_turns, max_turns = float('inf'), 0

    # a부터 b까지의 모든 숫자에 대해 이분 탐색을 수행합니다.
    for num in range(a, b + 1):
        turns = 0
        left, right = 1, m

        # 이분 탐색을 시작합니다.
        while left <= right:
            # 중간값을 구합니다.
            mid = (left + right) // 2
            turns += 1
            if mid == num:
                break
            elif mid < num:
                left = mid + 1
            else:
                right = mid - 1

        # 최소 횟수와 최대 횟수를 업데이트합니다.
        min_turns = min(min_turns, turns)
        max_turns = max(max_turns, turns)

    return min_turns, max_turns

m = int(input())
a, b = map(int, input().split())
min_duration, max_duration = game_duration(m, a, b)
print(min_duration, max_duration)

File: 241005/컴퓨터와 함께하는 숫자 게임 2/play-number-game-with-computer-2.cpp
#include <iostream>
#include <vector>
using namespace std;


vector<long long> arr; 
long long m; 

int binary_search(int target){
    int left = 0;
    int right = m-1;

    int time = 0;

    while(left <= right){
        time += 1;
        int mid = (left + right) / 2;

        if (arr[mid] == target)
            return time;
        if (arr[mid] > target)
            right = mid - 1;
        else
            left = mid + 1;

    }

}




int main() {
    // 여기에 코드를 작성해주세요.
    cin >> m; 

    for(long long i = 1; i <= m; i++){
        arr.push_back(i);
    }

    int a, b;
    cin >> a >> b;

    int max_t = 0;
    int min_t = 1000000; 
    for(long long i = a ; i <= b ; i++){
        int time = binary_search(i);
        max_t = max(max_t, time);
        min_t = min(min_t, time);
    }

    cout << min_t << " " << max_t ;



    return 0;
}